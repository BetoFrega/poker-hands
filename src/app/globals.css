:root {
  --background: #ffffff;
  --disabled: #adadad;
  --foreground: #171717;
}

@media (prefers-color-scheme: dark) {
  :root {
    --background: #0a0a0a;
    --disabled: #4b4b4b;
    --foreground: #ededed;
  }
}

body,
.docs-story {
  color: var(--foreground);
  background: var(--background);
  font-family: Arial, Helvetica, sans-serif;
}
