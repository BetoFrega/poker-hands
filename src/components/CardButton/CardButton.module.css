.CardButton {
  border: var(--foreground) 2px solid;
  border-radius: 5px;
  max-width: 100px;
  text-align: center;
  padding: 0.5em 0;
  & p {
    margin: 0;
    min-height: 1.15em;
  }
}

.pointer {
  cursor: pointer;
}

.empty {
  border-style: dotted;
}

.CardValue {
  font-size: 1em;
  @media (min-width: 920px) {
    font-size: 1.2em;
  }
}

.CardSuit {
  font-size: 1.8em;
  @media (min-width: 920px) {
    font-size: 2.4em;
  }
}

.selected {
  filter: brightness(0.5);
}
