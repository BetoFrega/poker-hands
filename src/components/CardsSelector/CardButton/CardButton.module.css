.CardButton {
  border: var(--foreground) 3px solid;
  border-radius: 2px;
  cursor: pointer;
  aspect-ratio: 3/4;
  max-width: 100px;
  text-align: center;
  padding: 1em 0;

  & p {
    margin: 0;
  }
}

.empty {
  border-style: dotted;
}

.CardValue {
  font-size: 2em;
}

.CardSuit {
  font-size: 2.5em;
}

.selected {
  filter: brightness(0.5);
}
