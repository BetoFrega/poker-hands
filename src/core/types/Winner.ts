export enum Winner {
  TIE,
  Player1,
  Player2,
}
