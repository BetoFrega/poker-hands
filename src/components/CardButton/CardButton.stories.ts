import { Meta, StoryObj } from "@storybook/react";
import { fn } from "@storybook/test";
import { CardSuitEnum, CardValueEnum } from "../../core/types/Card";
import { CardButton } from "./CardButton";

const deckCard = {
  card: {
    value: CardValueEnum.Queen,
    suit: CardSuitEnum.Hearts,
  },
};
const meta: Meta<typeof CardButton> = {
  component: CardButton,
  tags: ["autodocs"],

  args: {
    deckCard,
    onClick: fn(),
  },
};
export default meta;

type Story = StoryObj<typeof CardButton>;
export const Unselected: Story = {};
export const Selected: Story = {
  args: {
    deckCard: {
      ...deckCard,
      hand: 1,
    },
  },
};

export const Empty: Story = { args: { deckCard: null } };
