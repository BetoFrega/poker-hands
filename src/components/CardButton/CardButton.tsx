import { useCallback } from "react";
import { DeckCard } from "../../core/store/PokerStore";
import { Card, CardSuitEnum, CardValueEnum } from "../../core/types/Card";
import { cx } from "../../helpers/cx";
import styles from "./CardButton.module.css";

const suitIconMap: Record<CardSuitEnum, string> = {
  [CardSuitEnum.Hearts]: "♥",
  [CardSuitEnum.Clubs]: "♣",
  [CardSuitEnum.Diamonds]: "♦",
  [CardSuitEnum.Spades]: "♠",
};
const suitNameMap: Record<CardSuitEnum, string> = {
  [CardSuitEnum.Hearts]: "Hearts",
  [CardSuitEnum.Clubs]: "Clubs",
  [CardSuitEnum.Diamonds]: "Diamonds",
  [CardSuitEnum.Spades]: "Spades",
};
const valueNameMap: Record<CardValueEnum, string> = {
  [CardValueEnum.Two]: "Two",
  [CardValueEnum.Three]: "Three",
  [CardValueEnum.Four]: "Four",
  [CardValueEnum.Five]: "Five",
  [CardValueEnum.Six]: "Six",
  [CardValueEnum.Seven]: "Seven",
  [CardValueEnum.Eight]: "Eight",
  [CardValueEnum.Nine]: "Nine",
  [CardValueEnum.Ten]: "Ten",
  [CardValueEnum.Jack]: "Jack",
  [CardValueEnum.Queen]: "Queen",
  [CardValueEnum.King]: "King",
  [CardValueEnum.Ace]: "Ace",
};

export function CardButton({
  deckCard,
  onClick,
}: {
  deckCard: DeckCard | null;
  onClick?: (value?: Card) => void;
}) {
  const onClickHandler: () => void = useCallback(() => {
    onClick?.(deckCard?.card);
  }, [onClick, deckCard]);
  const cardName = deckCard
    ? `${valueNameMap[deckCard.card.value]} of ${suitNameMap[deckCard.card.suit]}`
    : "Empty card";
  return (
    <div
      className={cx([
        styles.CardButton,
        deckCard?.hand && styles.selected,
        deckCard === null && styles.empty,
        onClick && styles.pointer,
      ])}
      onClick={onClickHandler}
      aria-label={cardName}
      role="button"
    >
      <p className={styles.CardValue}>{deckCard?.card.value}</p>
      <p className={styles.CardSuit}>
        {deckCard && suitIconMap[deckCard.card.suit]}
      </p>
    </div>
  );
}
