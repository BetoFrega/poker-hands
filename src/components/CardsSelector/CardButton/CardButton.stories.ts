import { Meta, StoryObj } from "@storybook/react";
import { fn } from "@storybook/test";
import { CardSuitEnum, CardValueEnum } from "../../../core/types/Card";
import { CardButton } from "./CardButton";

const meta: Meta<typeof CardButton> = {
  component: CardButton,
  args: {
    card: {
      value: CardValueEnum.Queen,
      suit: CardSuitEnum.Hearts,
    },
    onClick: fn(),
  },
};
export default meta;

export const Unselected: StoryObj<typeof CardButton> = {};
export const Selected: StoryObj<typeof CardButton> = {
  args: { isSelected: true },
};
