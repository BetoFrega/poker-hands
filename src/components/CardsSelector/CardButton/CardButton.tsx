import { useCallback } from "react";
import { Card, CardSuitEnum } from "../../../core/types/Card";
import styles from "./CardButton.module.css";

const suitIconMap: Record<CardSuitEnum, string> = {
  [CardSuitEnum.Hearts]: "♥",
  [CardSuitEnum.Clubs]: "♣",
  [CardSuitEnum.Diamonds]: "♦",
  [CardSuitEnum.Spades]: "♠",
};

export function CardButton({
  card,
  isSelected,
  onClick,
}: {
  card: Card;
  onClick?: (value: Card) => void;
  isSelected?: boolean;
}) {
  const onClickHandler: () => void = useCallback(() => {
    onClick?.(card);
  }, [onClick, card]);
  return (
    <div
      className={`${styles.CardButton} ${isSelected ? styles.selected : ""}`}
      onClick={onClickHandler}
    >
      <p className={styles.CardValue}>{card.value}</p>
      <p className={styles.CardSuit}>{suitIconMap[card.suit]}</p>
    </div>
  );
}
