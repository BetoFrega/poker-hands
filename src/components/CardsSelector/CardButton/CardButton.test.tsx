import { describe, expect, it } from "@jest/globals";
import { render } from "@testing-library/react";
import { CardSuitEnum, CardValueEnum } from "../../../core/types/Card";
import { CardButton } from "./CardButton";

describe("CardButton", () => {
  describe("baseline", () => {
    it("should render unselected", () => {
      const { container } = render(
        <CardButton
          card={{ suit: CardSuitEnum.Hearts, value: CardValueEnum.Ace }}
          onClick={() => {}}
        />,
      );
      expect(container).toMatchSnapshot();
    });
    it("should render selected", () => {
      const { container } = render(
        <CardButton
          card={{ suit: CardSuitEnum.Hearts, value: CardValueEnum.Ace }}
          onClick={() => {}}
          isSelected
        />,
      );
      expect(container).toMatchSnapshot();
    });
  });
});
