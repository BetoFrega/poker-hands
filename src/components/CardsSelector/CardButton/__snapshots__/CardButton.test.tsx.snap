// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardButton baseline should render selected 1`] = `
<div>
  <div
    class="CardButton selected"
  >
    <p
      class="CardValue"
    >
      A
    </p>
    <p
      class="CardSuit"
    >
      ♥
    </p>
  </div>
</div>
`;

exports[`CardButton baseline should render unselected 1`] = `
<div>
  <div
    class="CardButton "
  >
    <p
      class="CardValue"
    >
      A
    </p>
    <p
      class="CardSuit"
    >
      ♥
    </p>
  </div>
</div>
`;
